bin/full-node/src/run/network_service.rs

```diff
@@ -586,13 +615,39 @@ impl NetworkService {
                 Duration::from_secs(12),
             );
 
+            // TODO: somehow cancel the request if the `rx` is dropped?
             guarded.blocks_requests.insert(request_id, tx);
 
             self.inner.wake_up_main_background_task.notify(1);
             rx
         };
 
-        rx.await.unwrap().map_err(BlocksRequestError::Request)
+        let result = rx.await.unwrap().map_err(BlocksRequestError::Request);
+
+        // Requet has finished. Print the log and prevent the cancellation message from being
+        // printed.
+        mem::forget(_log_if_cancel);
+        // TODO: print something if the request is cancelled?
```

Comment:
but we've just dropped the cancellation struct, why the need for todo?

Reply:
Mistake